packages/expo-notifications/src/NotificationPermissionsModule.ts

```diff
@@ -1,6 +1,73 @@
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+import { PermissionStatus } from 'unimodules-permissions-interface';
+
+import {
+  NativeNotificationPermissionsRequest,
+  NotificationPermissionsStatus,
+} from './NotificationPermissions.types';
 import { NotificationPermissionsModule } from './NotificationPermissionsModule.types';
 
+function convertPermissionStatus(
+  status?: NotificationPermission | 'prompt'
+): NotificationPermissionsStatus {
+  switch (status) {
+    case 'granted':
+      return {
+        status: PermissionStatus.GRANTED,
+        expires: 'never',
+        canAskAgain: false,
+        granted: true,
+      };
+    case 'denied':
+      return {
+        status: PermissionStatus.DENIED,
+        expires: 'never',
+        canAskAgain: false,
+        granted: false,
+      };
+    default:
+      return {
+        status: PermissionStatus.UNDETERMINED,
+        expires: 'never',
+        canAskAgain: true,
+        granted: false,
+      };
+  }
+}
+
+async function resolvePermissionAsync(shouldAsk: boolean): Promise<NotificationPermissionsStatus> {
```

Comment:
A nitpick, but could we make this a object-based argument? (Single `true`/`false` as an argument isn't obvious at first glance what it does. If it was `{ shouldAsk: true }` it'd be clear immediately.)